components/PostAddForm.js

```diff
@@ -0,0 +1,70 @@
+import { useRouter } from 'next/router';
+import { useState, useCallback } from 'react';
+import Textarea from 'react-textarea-autosize';
+
+const PostAddForm = () => {
+  const { query: { offset } } = useRouter();
+  const [isExpanded, setExpanded] = useState(false);
+  const handleFocus = useCallback(() => setExpanded(true), []);
+  const handleCancel = useCallback(() => setExpanded(false), []);
+
+  if (offset)  {
+    return false;
+  }
+
+  return (
+    <div>
+      <Textarea
+        minRows={3}
+        maxRows={10}
+        maxLength="1500"
+        defaultValue=""
+        onFocus={handleFocus}
+      />
```

Comment:
I think I would prefer to have a simple `<textarea>` here _until_ we need the full power of `<Textarea>`, i.e. until onFocus. This way, we don't need to instantiate the component for the 99% of time when we don't need it (people read their feeds much more than posting new entries).